packages/gestalt/src/OverlayPanel.js

```diff
@@ -116,6 +117,18 @@ function OverlayPanel({
   );
 }
 
+const DismissingElement = ({
```

Comment:
Nothing about this seems specific to OverlayPanel, so I think this is ripe for more abstraction. Consider something like this:
```js
// ./animation/DismissElement.js

type Props = {| children: ({| onDismissStart: () => void |}) => Node |};

export default function DismissElement({ children }: Props): Node {
  const { onExternalDismiss } = useAnimation();
  return children({ onDismissStart: onExternalDismiss });
}
```
```js
// OverlayPanel.js
import DismissElement from './animation/DismissElement.js';

OverlayPanel.DismissElement = DismissElement;
```
```js
// MobileSheet.js
import DismissElement from './animation/DismissElement.js';

MobileSheet.DismissElement = DismissElement;
```
etc